src/Activity.jsx

```diff
@@ -1,120 +1,356 @@
+    const annualData = data.filter(
+      val => val.createdAt.year() === now.getFullYear()
     );
+    result.forEach(day => {
+      let count = 0;
+      for (const activity of annualData) {
+        if (Number(day.x) === activity.createdAt.month() + 1) {
+          day.y += activity.avgDecibel;
+          count++;
+        }
+      }
+      if (count !== 0) {
+        day.y /= count;
+      }
+    });
+    this.drawBarChart(scale, result);
   };
 
-  renderChart = (xMax, xStepSize, yMax, yStepSize, points, avgdB) => {
-    const ctx = document.getElementById('line-chart');
+  drawBarChart = (scale, data) => {
+    const wrapper = document.getElementById('primaryChartAreaWrapper');
+    const canvas = document.getElementById('primaryChart');
+    canvas.remove();
+    const newCanvas = document.createElement('canvas');
+    newCanvas.setAttribute('id', 'primaryChart');
+    newCanvas.setAttribute('height', '300');
+    newCanvas.setAttribute('width', '1200');
+    wrapper.append(newCanvas);
+
+    const ctx = document.getElementById('primaryChart').getContext('2d');
     new Chart(ctx, {
-      type: 'line',
+      type: 'bar',
       data: {
-        datasets: [
-          {
-            data: points,
-            label: 'Recent',
-            borderColor: '#F9AA33',
-            fill: true,
-            backgroundColor: 'rgba(247, 168, 51,0.3)'
-          },
-          {
-            data: [{ x: 0, y: avgdB }, { x: xMax, y: avgdB }],
-            label: 'Avrage',
-            borderColor: '(83, 103, 114)',
-            fill: false
-          }
-        ]
+        labels: scale,
+        datasets: [{ data, backgroundColor: 'rgba(249, 170, 51, 0.6)' }]
       },
       options: {
         scales: {
-          xAxes: [
+          yAxes: [
             {
-              type: 'linear',
-              position: 'bottom',
-              ticks: {
-                min: 0,
-                max: xMax,
-                stepSize: xStepSize
-              }
+              ticks: { beginAtZero: true },
+              scaleLabel: { display: true, labelString: 'dB', fontSize: 18 }
             }
           ],
-          yAxes: [
+          xAxes: [
             {
-              ticks: {
-                min: 0,
-                max: yMax,
-                stepSize: yStepSize
-              }
+              barPercentage: 0.4,
+              ticks: { fontSize: 18 }
             }
           ]
         }
       }
     });
   };
 
+  handleTimeClick = event => {
+    const target = event.currentTarget;
+    this.setState({ timeAnchorEl: target });
+  };
+
+  handleActivityClick = event => {
+    const target = event.currentTarget;
+    this.setState({ activityAnchorEl: target });
+  };
+
+  handleTimeClose = e => {
+    const menuItem = e.currentTarget.textContent;
+    const {
+      activity,
+      projectionData,
+      karaokeData,
+      tongueTwisterData
+    } = this.state;
+
+    let data;
+    if (activity === 'Projection') data = projectionData;
```

Comment:
I recommend you to add {} even in single line condition because 1. that is better readable, 2. that avoids some problems on optimization code.